sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -681,10 +681,10 @@ class CastChecks extends ExprChecks {
   val booleanChecks: TypeSig = integral + fp + BOOLEAN + TIMESTAMP + STRING
   val sparkBooleanSig: TypeSig = numeric + BOOLEAN + TIMESTAMP + STRING
 
-  val integralChecks: TypeSig = integral + fp + BOOLEAN + TIMESTAMP + STRING + BINARY
+  val integralChecks: TypeSig = numeric + BOOLEAN + TIMESTAMP + STRING + BINARY
```

Comment:
You have an assert in the cast code indicating that it only works for ANSI casts, but here you have updated all casts to support decimal. We cannot merge this patch in like this as it will throw an exception instead of falling back to the CPU.

Reply:
I've updated `CastCheck` to support `ansiMode` check for numeric to decimal conversion.